### PR TITLE
Fix pairing and indexing issues

### LINKED ISSUE #49: Unpair volunteer and elderly befriendee contacts
As a voluntary organisation manager, I can unpair volunteer and elderly befriendee contacts, so that I can delete obsolete contacts, easily keep track of which volunteers are paired with which elderly befriendees, and relay relevant information accordingly.

### LINKED ISSUE #55: Update storage with pairedWith attribute
Update the storage such that the pairedWith attribute is stored and updated accordingly.

### PR DESCRIPTION
**Pull request by chaaaaun**
Issues resolved:
- Display list index for each list
- Update all commands except delete to index based on split lists
- Persist pair data to storage

### CODE REVIEW

**Review by sdevih — APPROVED** [review]
LGTM, test cases to be fixed after merging.

**Review by CJerrong — APPROVED** [review]
LGTM, just gotta refactor the test cases

**Pull request merged**
